pkgs/development/python-modules/pleroma-bot/default.nix

```diff
@@ -0,0 +1,28 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, oauthlib
+, requests_oauthlib
+, requests
+, pyaml
+}:
+
+buildPythonPackage rec {
+  pname = "pleroma-bot";
+  version = "0.8.6";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "85056df457a49f2adbd36349d821c9401003d91a753de6456ceb805770e38c13";
+  };
+
+  doCheck = false;
```

Comment:
The upstream repo contains [`tests`](https://github.com/robertoszek/pleroma-bot/tree/master/pleroma_bot/tests). If they are mocking network access properly, perhaps we could run them.

If it's not possible to run the tests, at least a `pythonImportsCheck` should be added.

Reply:
The archive retrieved by ```fetchPypi``` doesn't include ```tests```, not sure how to go about that.
Would it be possible and/or acceptable to retrieve them some other way in order to run them? (checking the documentation I've not come across any example or mention of retrieving tests from a different source than the package itself).

I've added ```pythonImportsCheck``` in the meantime, as requested.

Reply:
Please fetch the repo from github or the hosting platform of their choice instead.

Reply:
Understood, I changed it to ```fetchFromGitHub``` and added the pytest tests. 
The package builds successfully and the tests pass.